Sources_v3/StreamChatUI/ChatChannel/MessageComposer/MessageComposerViewController.swift

```diff
@@ -358,50 +379,71 @@ open class MessageComposerViewController<ExtraData: ExtraDataTypes>: ViewControl
     }
     
     // MARK: UITextView
-    
+
     @objc func promptSuggestionIfNeeded(for text: String) {
-        // Check if first symbol is `/` and if there are available commands in `ChatConfig`.
-        guard text.trimmingCharacters(in: .whitespacesAndNewlines).first == "/",
-            let commands = controller?.channel?.config.commands
-        else {
+        if text.trimmingCharacters(in: .whitespacesAndNewlines).contains("@") {
+            if let index = (text.range(of: "@")?.upperBound) {
+                let textAfterAtSymbol = String(text.suffix(from: index))
+                promptMentions(for: textAfterAtSymbol)
+            }
+
+        } else if let commands = controller?.channel?.config.commands,
+            text.trimmingCharacters(in: .whitespacesAndNewlines).first == "/" {
+            prompt(commands: commands, for: text)
+        } else {
             dismissSuggestionsViewController()
-            return
         }
-        
+    }
+
+    func prompt(commands: [Command], for text: String) {
         // Get the command value without the `/`
         let typedCommand = String(text.trimmingCharacters(in: .whitespacesAndNewlines).dropFirst())
-        
+
         // Set all commands as hints initially
         var commandHints: [Command] = commands
-        
+
         // Filter commands when user is typing something after `/`
         if !typedCommand.isEmpty {
             commandHints = commands.filter { $0.name.range(of: typedCommand, options: .caseInsensitive) != nil }
         }
 
         showSuggestionsViewController(
-            with: .commands(commandHints),
-            onSelectItem: { [weak self] index in
-                self?.state = .slashCommand(commandHints[index])
+            for: .command(hints: commandHints),
+            onSelectItem: { [weak self] commandIndex in
+                self?.state = .slashCommand(commandHints[commandIndex])
             }
         )
     }
-    
+
+    func promptMentions(for text: String) {
+        userSuggestionSearchController.search(term: text)
+
+        showSuggestionsViewController(
+            for: .mention,
+            onSelectItem: { [weak self] _ in
+                print(text)
+                print(self?.textView.text)
+                self?.textView.text = self?.textView.text.appending("@ \(text)")
+                self?.state = .initial
+            }
+        )
```

Comment:
This probably shouldn't be called every time the search term changes. You need to present the VC just once